lib/matplotlib/legend.py

```diff
@@ -983,6 +983,12 @@ def _auto_legend_data(self):
                 if len(hoffsets):
                     for offset in transOffset.transform(hoffsets):
                         offsets.append(offset)
+                if isinstance(artist, PolyCollection):
+                    paths = artist.get_paths()
+                    for path in paths:
+                        lines.append(artist.get_transform().transform_path(path))
```

Comment:
I would probably just have this in the if/elif chain rather than inside of one of the cases.